### PR TITLE
Remove references to missions that do not exist.

### LINKED ISSUE #32366: src/generic_factory.h:355 [const T& generic_factory<T>::obj(const string_id<T>&) const [with T = mission_type]] invalid mission_type id "MISSION_REFUGEE_John_1"
# Describe the bug

I was reaching the refugee center, and then
```
	Attempting to repeat stack trace using debug symbols...
	debug_write_backtrace(std::ostream&)
	.../src/debug.cpp:621
	DebugLog(DebugLevel, DebugClass)
	.../src/debug.cpp:786
	realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
	.../src/debug.cpp:108
	std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()
	/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/bits/basic_string.h:657
	void realDebugmsg<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char const*>(char const*, char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char const*&&)
	.../src/debug.h:79
	generic_factory<mission_type>::obj(string_id<mission_type> const&) const
	.../src/generic_factory.h:355
	generic_factory<mission_type>::obj(string_id<mission_type> const&) const
	.../src/generic_factory.h:352
	string_id<mission_type>::obj() const
	.../src/missiondef.cpp:187
	mission::reserve_new(string_id<mission_type> const&, int)
	.../src/mission.cpp:61
	npc::add_new_mission(mission*)
	.../src/npc.cpp:2170 (discriminator 1)
	npc::load_npc_template(string_id<npc_template> const&)
	.../src/npc.cpp:289 (discriminator 1)
	map::place_npc(int, int, string_id<npc_template> const&, bool)
	.../src/mapgen.cpp:6940
	jmapgen_npc::apply(mapgendata const&, jmapgen_int const&, jmapgen_int const&, float, mission*) const
	.../src/mapgen.cpp:790
	jmapgen_objects::apply(mapgendata const&, float, mission*) const
	.../src/mapgen.cpp:2649
	jmapgen_objects::apply(mapgendata const&, int, int, float, mission*) const
	.../src/mapgen.cpp:2661
	mapgen_function_json::generate(map*, int_id<oter_t> const&, mapgendata const&, time_point const&, float)
	.../src/mapgen.cpp:2614
	run_mapgen_func(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, map*, int_id<oter_t>, mapgendata, time_point const&, float)
	.../src/mapgen.cpp:8512
	map::draw_map(int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, int_id<oter_t> const&, time_point const&, float, int, regional_settings const*)
	.../src/mapgen.cpp:2705 (discriminator 1)
	int_id<oter_t>::operator->() const
	.../src/int_id.h:97
	map::generate(int, int, int, time_point const&)
	.../src/mapgen.cpp:151
	map::loadn(int, int, int, bool)
	.../src/map.cpp:6763 (discriminator 3)
	map::shift(int, int)
	.../src/map.cpp:6566 (discriminator 2)
	game::update_map(int&, int&)
	.../src/game.cpp:10445
	game::update_map(player&)
	.../src/game.cpp:10410
	map::displace_vehicle(tripoint&, tripoint const&)
	.../src/map.cpp:1104
	map::move_vehicle(vehicle&, tripoint const&, tileray const&)
	.../src/map.cpp:594
Backtrace emission took 1 seconds.
src/generic_factory.h:355 [const T& generic_factory<T>::obj(const string_id<T>&) const [with T = mission_type]] invalid mission_type id "MISSION_REFUGEE_John_1"
```

070b9e9606f6063dd1df00d071b272d8f57ce3c9

### PR DESCRIPTION
**Pull request by BevapDin**
```Summary: None```

Fixes #32366.

@I-am-Erk may have the missing mission definitions somewhere as they introduced them in #31290. If so, add them and revert this commit.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
I had those added in to circumvent an issue where, when NPCs get new missions added, they need to be fully regenerated from scratch. The intent is that as the center gets expanded it won't be necessary to find a new center every time there's a PR to add more missions.

**Comment by BevapDin** [comment]
Well, having references to non-existing things causes the errors as mentioned in the linked issue.

Do we want those errors?

If not, the references to those missions need to be removed.

**Comment by I-am-Erk** [comment]
I suppose it's reasonable to take them out for now, I can re-add them with dummy missions when I have time but I have no idea when that'll be.

**Comment by BevapDin** [comment]
> I had those added in to circumvent an issue where, when NPCs get new missions added, they need to be fully regenerated from scratch. The intent is that as the center gets expanded it won't be necessary to find a new center every time there's a PR to add more missions.

It does not even work like you intended it. The game loads the NPC data as template. When an actual NPC is spawned (e.g. when the player triggers generating the map of the center), a real NPC is created based on that template. The mission type is used at *that* point to create a matching mission and assign it to the new NPC. From there it's fixed: the NPC exists, the mission exists. Changes in the NPC template won't affect already generated NPCs. The player will have to trigger the generation of a new NPC to get new mission types.

Before the NPC is created, the mission type id is loaded, but when the game ends, it's forgotten anyway.

----

In short: NPCs need to be created from scratch to use your new mission definitions. Storing the mission type id of your planned mission has no effect at all.

**Comment by I-am-Erk** [comment]
That's unfortunate. It's a behaviour I'd love to change, but certainly not high on my list of priorities.
